### PR TITLE
feat/conversation app api

### PR DESCRIPTION
**Pull request by Dovchik**
- Covers [App](https://developers.sinch.com/docs/conversation/api-reference/conversation/tag/App/) endpoints.
- Http refactor.
- Typos

### CODE REVIEW

File: src/Sinch/Conversation/Apps/Credentials/BasicAuthCredentials.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        /// <summary>
+        ///     Returns the string presentation of the object
+        /// </summary>
+        /// <returns>String presentation of the object</returns>
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+            sb.Append("class BasicAuthCredential {\n");
+            sb.Append("  Password: ").Append(Password).Append("\n");
```
Comment (JPPortier) [inline src/Sinch/Conversation/Apps/Credentials/BasicAuthCredentials.cs:37]: Should we offer this critical information ?
This mean we could found clear password in any layer using a print/tostring onto such instance (log files, ...)

File: src/Sinch/Core/Utils.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ public static T ParseEnum<T>(string value)
                 }
             }
 
-            throw new InvalidOperationException($"Failed to parse {nameof(T)} enum");
+            throw new InvalidOperationException($"Failed to parse {enumType.Name} enum");
```
Comment (JPPortier) [inline src/Sinch/Core/Utils.cs:75]: Does this mean any unknown value will trigger an exception ?
My thoughts from https://github.com/sinch/sinch-sdk-internal-docs/blob/main/ADR/0005-enums.md was we should be able to handle new values *dynamically*: 
> User is able to create new values if needed.
Deserialization is able to create new instances if needed.

Currently, we will have different behavior between `.net` and `java` SDKs (https://github.com/sinch/sinch-sdk-java-proto/pull/2)

File: src/Sinch/Conversation/Apps/Credentials/BasicAuthCredentials.cs (hunk 1 above)
Comment (Dovchik) [reply src/Sinch/Conversation/Apps/Credentials/BasicAuthCredentials.cs:37]: Good point. Will be removed.

File: src/Sinch/Core/Utils.cs (hunk 2 above)
Comment (Dovchik) [reply src/Sinch/Core/Utils.cs:75]: Yeah, this is intended. Currently I want it to be left as it is. We can revisit it latter nonetheless.

**Review by JPPortier — APPROVED** [review]

**Review by 650elx — APPROVED** [review]

**Pull request merged**